src/components/Campaign.tsx

```diff
@@ -52,11 +52,23 @@ export const Campaign = () => {
     campaignContentMutation.mutate({ idCampaign, content });
   };
 
-  const nextUrl =
-    searchParams.get("redirectedFromSummary")?.toUpperCase() === "TRUE"
-      ? `${dopplerLegacyBaseUrl}/Campaigns/Summary/Index?IdCampaign=${idCampaign}`
-      : `${dopplerLegacyBaseUrl}/Campaigns/Recipients/Index?IdCampaign=${idCampaign}` +
-        `&RedirectedFromSummary=False&RedirectedFromTemplateList=False`;
+  const getButtonUrls = (buttonRef: string) => {
+    if (searchParams.get("redirectedFromSummary")?.toUpperCase() === "TRUE") {
+      return `${dopplerLegacyBaseUrl}/Campaigns/Summary/Index?IdCampaign=${idCampaign}`;
+    } else {
+      if (buttonRef === "nextUrl") {
+        return (
+          `${dopplerLegacyBaseUrl}/Campaigns/Recipients/Index?IdCampaign=${idCampaign}` +
+          `&RedirectedFromSummary=False&RedirectedFromTemplateList=False`
+        );
+      } else {
+        return (
+          `${dopplerLegacyBaseUrl}/Campaigns/Content?IdCampaign=${idCampaign}` +
+          `&RedirectedFromSummary=False&RedirectedFromTemplateList=False&RedirectedFromCampaignB=False`
+        );
+      }
+    }
+  };
```

Comment:
I think that there are ways to get this code clear. 

One could be use `else if` to reduce nesting:

```suggestion
  const getButtonUrls = (buttonRef: string) => {
    if (searchParams.get("redirectedFromSummary")?.toUpperCase() === "TRUE") {
      return `${dopplerLegacyBaseUrl}/Campaigns/Summary/Index?IdCampaign=${idCampaign}`;
    } else if (buttonRef === "nextUrl") {
      return `${dopplerLegacyBaseUrl}/Campaigns/Recipients/Index?IdCampaign=${idCampaign}` +
        `&RedirectedFromSummary=False&RedirectedFromTemplateList=False`;
    } else {
      return `${dopplerLegacyBaseUrl}/Campaigns/Content?IdCampaign=${idCampaign}` +
          `&RedirectedFromSummary=False&RedirectedFromTemplateList=False&RedirectedFromCampaignB=False`;
    }
  };
```

But I still do not like it. Because:

* We are calling the function twice with dow different parameters, but both results are used.
* Having a string parameter is error-prone. There is another mobile piece in the code.
* There is another abstraction, so, it is difficult to understand it at a glance.

Maybe something like this:

```suggestion
  const redirectedFromSummary = searchParams.get("redirectedFromSummary")?.toUpperCase() === "TRUE";
  const nextUrl = redirectedFromSummary
      ? `${dopplerLegacyBaseUrl}/Campaigns/Summary/Index?IdCampaign=${idCampaign}`
      : `${dopplerLegacyBaseUrl}/Campaigns/Recipients/Index?IdCampaign=${idCampaign}` +
        `&RedirectedFromSummary=False&RedirectedFromTemplateList=False`;
        
  const backUrl = redirectedFromSummary
      ? `${dopplerLegacyBaseUrl}/Campaigns/Summary/Index?IdCampaign=${idCampaign}`
      : `${dopplerLegacyBaseUrl}/Campaigns/Content?IdCampaign=${idCampaign}` +
          `&RedirectedFromSummary=False&RedirectedFromTemplateList=False&RedirectedFromCampaignB=False`;
```

Or

```suggestion
  const redirectedFromSummary = searchParams.get("redirectedFromSummary")?.toUpperCase() === "TRUE";
  const summaryUrl = `${dopplerLegacyBaseUrl}/Campaigns/Summary/Index?IdCampaign=${idCampaign}`;
  const nextStepUrl = `${dopplerLegacyBaseUrl}/Campaigns/Recipients/Index?IdCampaign=${idCampaign}` +
    `&RedirectedFromSummary=False&RedirectedFromTemplateList=False`;
  const backStepUrl = `${dopplerLegacyBaseUrl}/Campaigns/Content?IdCampaign=${idCampaign}` +
          `&RedirectedFromSummary=False&RedirectedFromTemplateList=False&RedirectedFromCampaignB=False`;

  const nextButtonUrl = redirectedFromSummary ? summaryUrl : nextStepUrl;
  const backButtonUrl = redirectedFromSummary ? summaryUrl : backStepUrl;
```

Reply:
Fixed

Reply:
Thanks!